### PR TITLE
Remove dependency on deprecated `DkgPublicParams`

### PR DESCRIPTION
**Pull request by piotr-roslaniec**
**Type of PR:**
- Other

**Required reviews:** 
-  2

**What this does:**
- Removes dependency on `DkgPublicParams` which is now replaced with a default value
- Introduces `ValidatorMessage` type where necessary

**Why it's needed:**
- To implement downstream changes from https://github.com/nucypher/nucypher-core/issues/62

**Notes for reviewers:**
- Uses an [unreleased `ferveo` version](https://github.com/nucypher/ferveo/pull/127/commits/0d4e973e007b16cff34d649ae107608c809349af)

### CODE REVIEW

File: nucypher/crypto/ferveo/dkg.py [hunk 1]
```diff
@@ -51,21 +51,20 @@ def derive_public_key(*args, **kwargs) -> DkgPublicKey:
 
 def aggregate_transcripts(
     transcripts: List[Tuple[Validator, Transcript]], shares: int, *args, **kwargs
-) -> Tuple[AggregatedTranscript, DkgPublicKey, DkgPublicParameters]:
+) -> Tuple[AggregatedTranscript, DkgPublicKey]:
```
[the remark this answers is not in the document]
Comment (piotr-roslaniec) [reply nucypher/crypto/ferveo/dkg.py:54]: I was considering things but there was some [logic dependent on the `Transcript` being `None`](https://github.com/nucypher/nucypher/blob/44d1e81b8fad0287a7447606847ec0f31bb59db9/nucypher/blockchain/eth/actors.py#L380) in some cases and I decided to hold back, but I'm open to redoing this logic.

**Review by theref — APPROVED** [review]
Looks good!

File: nucypher/crypto/ferveo/dkg.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (derekpierre) [reply nucypher/crypto/ferveo/dkg.py:54]: Interesting. Is `None` allowed as a `Transcript` value for `ValidatorMessage`?

[the remark this answers is not in the document]
Comment (piotr-roslaniec) [reply nucypher/crypto/ferveo/dkg.py:54]: If we care about failures in some cases, perhaps we should return `msgs, failures = self._resolve_validators(ritual)`, where `failures` are `validator`-s who failed to post their `transcript`

[the remark this answers is not in the document]
Comment (derekpierre) [reply nucypher/crypto/ferveo/dkg.py:54]: Interesting. Maybe not `failures` per se but really "validators with no transcript"; This is case 1 above.

In any case, if you think about it and a clean solution comes out, great. If a clean solution doesn't come out of it, then no worries, and you can leave the code as is.

**Review by KPrasch — APPROVED** [review]
👨🏻‍🚀

**Pull request merged**